safe/impact_functions/generic/classified_polygon_people/impact_function.py

```diff
@@ -0,0 +1,411 @@
+
+        all_affected_geometry = []
+
+        all_areas = {}
+        imp_areas = {}
+
+        for f in exposure.getFeatures(
+                QgsFeatureRequest(extent_exposure)):
+            geometry = f.geometry()
+
+            if geometry is not None:
+                bbox = geometry.boundingBox()
+                geometry_area = geometry.area()
+            else:
+                continue
+
+            # clip the exposure geometry to requested extent if necessary
+            if not extent_exposure.contains(bbox):
+                geometry = geometry.intersection(extent_exposure_geom)
+
+            area_type = f[type_attr]
+            area_id = f.attribute(area_id_attribute)
+            self.all_areas_population[area_id] = f.attribute(
+                area_population_attribute)
+
+            # add to the total area of this land cover type
+            if area_type not in all_areas:
+                all_areas[area_type] = 0.
+
+            if area_id not in self.all_areas_ids:
+                self.all_areas_ids[area_id] = 0.
+
+            area = geometry.area()
+
+            all_areas[area_type] += area
+            self.all_areas_ids[area_id] += geometry_area
+
+            # find possible intersections with hazard layer
+            impacted_geometries = []
+            # unaffected_geometries = []
+            # impacted_features = {}
+            for hazard_id in hazard_index.intersects(bbox):
+                hazard_geometry = hazard_features[hazard_id].geometry()
+                impact_geometry = geometry.intersection(hazard_geometry)
+
+                if not impact_geometry.wkbType() == QGis.WKBPolygon and \
+                   not impact_geometry.wkbType() == QGis.WKBMultiPolygon:
+                    continue  # no intersection found
+
+                # find unaffected area geometry
+                unaffected_geometry = geometry.symDifference(impact_geometry)
+
+                all_affected_geometry.append(impact_geometry)
+
+                # add to the affected area of this area type
+                if area_type not in imp_areas:
+                    imp_areas[area_type] = 0.
+                if area_id not in self.all_affected_areas:
+                    self.all_affected_areas[area_id] = 0.
+                area = impact_geometry.area()
+                imp_areas[area_type] += area
+                self.all_affected_areas[area_id] += area
+
+                self.assign_impact_level(f,
+                                         hazard_id,
+                                         hazard_features,
+                                         unaffected_fields,
+                                         impact_fields,
+                                         unaffected_geometry,
+                                         impact_geometry,
+                                         writer)
+
+                impacted_geometries.append(impact_geometry)
+
+        return impacted_geometries
```

Comment:
If `exposure.getFeatures(QgsFeatureRequest(extent_exposure))` has 0 feature, this will raise an error, since the `impacted_geometries` hasn't been defined.